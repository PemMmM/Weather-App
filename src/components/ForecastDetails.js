import React from "react";
import PropTypes from "prop-types";
// import moment from "moment";
// import WeatherIcon from "react-icons-weather";

function ForecastDetails({ forecast }) {
  const { temperature, wind, humidity } = forecast;

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__temperature">
        {temperature.min}
        &deg;C
      </div>

      <div className="forecast-details__wind">
        {(wind.speed, wind.direction)}
      </div>
      <div className="forecast-details__humidity"> {humidity}</div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
    temperature: PropTypes.shape({
      min: PropTypes.number,
    }),
  }).isRequired,
};

export default ForecastDetails;
