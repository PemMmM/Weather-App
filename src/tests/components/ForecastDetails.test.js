import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("Forecast Details", () => {
  const validProps = {
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 10,
      direction: "s",
    },
    humidity: 30,
  };
  it("it renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment).toMatchSnapshot();
  });
  it("render correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);
    expect(getByText(/30/g)).toHaveClass("forecast-details__humidity");
    expect(getByText(/12/g)).toHaveClass("forecast-details__temperature");
  });
});
