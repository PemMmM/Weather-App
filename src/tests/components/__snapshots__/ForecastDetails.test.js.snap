// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details it renders correctly 1`] = `[Function]`;
